src/printer/smt2/smt2_printer.cpp

```diff
@@ -73,14 +73,22 @@ static void toStreamRational(std::ostream& out,
                              Variant v)
 {
   bool neg = r.sgn() < 0;
+  bool arithTokens = options::ioutils::getPrintArithLitToken(out);
   // Print the rational, possibly as a real.
   // Notice that we print (/ (- 5) 3) instead of (- (/ 5 3)),
   // the former is compliant with real values in the smt lib standard.
   if (r.isIntegral())
   {
     if (neg)
     {
-      out << "(- " << -r;
+      if (arithTokens)
```

Comment:
Too many `arithTokens` checks. Could you reduce them to just 2 checks: one for integers and one for rationals?

Reply:
Sure, this is a bit more verbose but I've updated now.